### ISSUE TITLE
setup MQTT server 

### ISSUE THREAD

**Issue by fluxrock** [flaviut/emporia-vue2-reversing#5]
Hello!
Could you please explain how did you setup the local network and bridge MQTT server with 192.168.1.101  address for getting the prod/minions/emporia/ct/v1/AAAABBBBCCCCDDDDEEEEFF/debug/v2 topic ?
Thank you!

**Comment by flaviut** [comment]
Unfortunately, it depends heavily on the exact details of how your home network is set up. I used linux and hostapd, but I never went down this path fully. I ended up discovering how to modify the flash image so that it'd use my wifi network, which made doing so pointless.

I'd recommend following the instructions at https://gist.github.com/flaviut/93a1212c7b165c7674693a45ad52c512 to run ESPHome.

**Comment by fluxrock** [comment]
Great work, Congratulations !
I'll do the  espHome route!, and keep looking the non soldering way too, if lucky find the ota magic byte or something else.
By the way, if change the firmware to espHome, is it possible to put the original firmware back ?
Also if you are interested, i can write about the other method to putting the esp32 in download mode, without using the VCC pin at all, it involves grounding the enable and io1 like you mention in your blog.

![emporiavueserial](https://user-images.githubusercontent.com/67212264/149864081-adbd8376-a803-4a49-ada7-6a2900ca5b52.jpg)

**Comment by fluxrock** [comment]
Great work, Congratulations !
I'll do the espHome way!  and keep looking the non soldering way too, if lucky find the ota magic byte or something else.
By the way, if change the firmware to espHome, is it possible to put the original firmware back ?
Also is there a special reason you change the other method to putting the esp32 in download mode? without using the VCC pin at all,  grounding the enable and io1 like you mention in your blog.

![emporiavueserial](https://user-images.githubusercontent.com/67212264/149866152-89bf10e9-150e-469a-be61-a9e9597fbe2c.jpg)

**Comment by flaviut** [comment]
Source code for the framework they use can be found here: https://github.com/aws/aws-iot-device-sdk-embedded-C. OTA would be really nice, but Amazon generally knows what they're doing security-wise (tuya… maybe not so much. but even they caught on eventually).

> By the way, if change the firmware to espHome, is it possible to put the original firmware back ?

Absolutely. That's why the first step in my guide is to back up the existing firmware. You can just keep that file around, and just flash it on if you want to switch back.

> Also if you are interested, i can write about the other method to putting the esp32 in download mode, without using the VCC pin at all, it involves grounding the enable and io1 like you mention in your blog.

Yup, but I'd rather keep it simple. I don't want to confuse anyone by giving more choices than they need.

Very cool idea bringing the serial wires out through there. Definitely seems nice for testing.

**Comment by flaviut** [comment]
> Also is there a special reason you change the other method to putting the esp32 in download mode? without using the VCC pin at all, grounding the enable and io1 like you mention in your blog. 

Yes. The original revision of the guide suggested that you hold the pins in the hole without soldering a header. In that situation, I think it's a little easier to do things with the VCC pin.

However, I discovered that 90%+ of boards these days are coming with the holes filled with solder. So I removed that part of the guide, since it's no longer useful. But I didn't change to EN/IO0 because it doesn't really matter.

**Comment by fluxrock** [comment]
aww, AWS seems like a long shot...
Is good to know the backup can go back in, I feel more confident to try the espHome.
True, my emporia had the holes filled in as well, I ended up not using header pins and just soldered the wires using those fills.
Thank you for the information and congratulations again for such a great work !

**Comment by josesa** [comment]
> Great work, Congratulations ! I'll do the espHome way! and keep looking the non soldering way too, if lucky find the ota magic byte or something else. By the way, if change the firmware to espHome, is it possible to put the original firmware back ? Also is there a special reason you change the other method to putting the esp32 in download mode? without using the VCC pin at all, grounding the enable and io1 like you mention in your blog.
> 
> ![emporiavueserial](https://user-images.githubusercontent.com/67212264/149866152-89bf10e9-150e-469a-be61-a9e9597fbe2c.jpg)

I know this has a long time, but worth the shot:

Did you have any issue with connecting the 2 grounds together? The USB/FTDI and the emporia (mains supplied)? 
I wonder how I can get the Serial output of the device in normal operation without frying the 2 different power supplies...
